### PR TITLE
cpio: don't try to hard link directory

### PR DESCRIPTION
**Pull request by fhs**
Signed-off-by: Fazlul Shahriar <fshahriar@gmail.com>

### CODE REVIEW

**Review by rminnich — APPROVED** [review]

**Comment by rminnich** [comment]
the part I don't understand is why this ever worked.

File: cmds/core/cpio/cpio.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func main() {
 
 			// A file with zero size could be a hard link to another file
 			// in the archive. The file with contents always comes first.
-			if rec.Info.FileSize == 0 {
+			if rec.Mode&cpio.S_IFMT != cpio.S_IFDIR && rec.Info.FileSize == 0 {
```
Comment (gmacd) [inline cmds/core/cpio/cpio.go:85]: Could use rec.Info.IsDir?

**Review by gmacd — DISMISSED** [review]

File: cmds/core/cpio/cpio.go (hunk 1 above)
Comment (fhs) [reply cmds/core/cpio/cpio.go:85]: There is no `IsDir` method. I can add it if you want.

**Comment by fhs** [comment]
> the part I don't understand is why this ever worked.

It works if the directories have different inode, but uroot seems to generate archives with all inodes set to 0.

File: cmds/core/cpio/cpio.go (hunk 1 above)
Comment (gmacd) [reply cmds/core/cpio/cpio.go:85]: No, my mistake

**Review by gmacd — APPROVED** [review]

**Pull request merged**